client/web/compose/src/components/ModuleFields/Editor/Geometry.vue

```diff
@@ -156,75 +156,30 @@
         {{ $t('clickToPlaceMarker') }}
       </template>
 
-      <div
-        v-if="!field.options.hideGeoSearch"
-        class="geosearch-container"
-      >
-        <c-input-search
-          v-model="geoSearch.query"
-          :placeholder="$t('geosearchInputPlaceholder')"
-          :autocomplete="'off'"
-          :debounce="300"
-          @input="onGeoSearch"
-        />
-
-        <div class="geosearch-results">
-          <div
-            v-for="(result, idx) in geoSearch.results"
-            :key="idx"
-            class="geosearch-result"
-            @click="placeGeoSearchMarker(result)"
-          >
-            {{ result.label }}
-          </div>
-        </div>
-      </div>
-
-      <l-map
-        ref="map"
-        :zoom="map.zoom"
-        :center="map.center"
-        style="height: 75vh; width: 100%; cursor: pointer;"
-        @click="placeMarker"
-        @locationfound="onLocationFound"
-      >
-        <l-tile-layer
-          url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
-          :attribution="map.attribution"
-        />
-        <l-marker
-          v-for="(marker, i) in markers"
-          :key="i"
-          :lat-lng="marker"
-          :opacity="localValueIndex === undefined || i == localValueIndex ? 1.0 : 0.6"
-          @click="removeMarker(i)"
-        />
-        <l-control class="leaflet-bar">
-          <a
-            v-if="!field.options.hideCurrentLocationButton"
-            :title="$t('tooltip.goToCurrentLocation')"
-            role="button"
-            class="d-flex justify-content-center align-items-center"
-            @click="goToCurrentLocation"
-          >
-            <font-awesome-icon
-              :icon="['fas', 'location-arrow']"
-              class="text-primary"
-            />
-          </a>
-        </l-control>
-      </l-map>
+      <c-map
+        ref="c-map"
+        :field="field"
+        :map="map"
+        :hide-geo-search="!field.options.hideGeoSearch"
+        :hide-current-location-button="field.options.hideCurrentLocationButton"
+        :active-marker-index="localValueIndex"
+        :markers="markers"
+        :labels="{
+          tooltip: { 'goToCurrentLocation': $t('tooltip.goToCurrentLocation') }
+        }"
+        @on-map-click="placeMarker"
+        @on-marker-marker="removeMarker"
+        @trigger-geosearch-error="triggerGeoSearchError"
+      />
     </b-modal>
   </b-form-group>
 </template>
 <script>
 import base from './base'
 import { latLng } from 'leaflet'
```

Comment:
Is there any way we can do this inside the CMap, if so we then can remove all imports from the webapp relating to the map